include/deal.II/lac/petsc_sparse_matrix.h

```diff
@@ -498,6 +501,24 @@ namespace PETScWrappers
       void
       reinit(const SparseMatrix &other);
 
+      /**
+       * Create a matrix where the size of the IndexSets determine the
+       * global number of rows and columns and the entries of the IndexSet
+       * give the rows and columns for the calling processor. Note that only
+       * ascending, 1:1 IndexSets are supported. The additional call to the
+       * local to global mappings is required to create the matrix of type
+       * IS (see DoFTools::extract_locally_active_dofs).
+       * This is required by the BDDC preconditioner.
+       */
+      template <typename SparsityPatternType>
+      void
+      reinit_IS(const IndexSet &           local_rows,
```

Comment:
```suggestion
      reinit(const IndexSet &           local_rows,
```
The other functions also take index sets so lets just reuse the same name - our convention is that we always call this `reinit()`.

Reply:
Ok. Is there a good place to put some kind of warning due to the different underlying matrix type? 

If not, it is always possible to just assume that the user knows what they are doing. Surely not always the case for me 😅, that's why I kept it separate.

Reply:
The right answer here is to document the requirement and use some combination of `MatGetType()` and our own `AssertThrow()` macro to guarantee that the correct `Mat` type is present.